### PR TITLE
Add TraceFunc

### PR DESCRIPTION
**Pull request by bmarini**
``` go
func MyFunc(ctx context.Context) {
    t := newrelic.TraceFunc(ctx)
    defer t.End()
}
```

### CODE REVIEW

File: tx.go [hunk 1]
```diff
@@ -176,6 +178,21 @@ func TraceGeneric(ctx context.Context, name string) *Trace {
 	})
 }
 
+// TraceDatastore adds a datastore segment to the newrelic transaction, if one exists in the context.
+func TraceDatastore(ctx context.Context, table, operation, sql, rollupName string) *Trace {
```
Comment (v-yarotsky) [inline tx.go]: `sql` is raw sql query string? What is `rollupName`?

Comment (bmarini) [inline tx.go]: Yep, sql is the raw query string. The agent obfuscates the sql. I'm not sure what the rollupName is, but I assume is allows you to group similar queries together or something.

**Comment by v-yarotsky** [comment]
:+1:

File: tx.go [hunk 2]
```diff
@@ -191,6 +208,18 @@ func trace(ctx context.Context, name string, fn func(Tx) error) *Trace {
 	return &Trace{nil, func() error { return nil }}
 }
 
+// caller returns the name of the function that called the function this function was called from.
+// n = 1 => caller of caller()
+// n = 2 => caller of caller of call()
+// etc.
+func caller(n int) string {
```
Comment (sanjayprabhu) [inline tx.go]: We already have this in `sdk.go`. Think it's worth extracting out?

**Comment by sanjayprabhu** [comment]
:+1:

File: tx.go (hunk 2 above)
Comment (bmarini) [inline tx.go]: I thought about it. Kind of a wierd func to export from sdk though.

**Pull request merged**